qa/tasks/cephadm.py

```diff
@@ -67,6 +68,29 @@ def build_initial_config(ctx, config):
     return conf
 
 
+def _get_trusted_ip_list(ctx, cluster_name):
+
+    trusted_ip_list = []
+    for remote, roles in ctx.cluster.remotes.items():
+        for role in [r for r in roles
+                     if teuthology.is_type('iscsi', cluster_name)(r)]:
+            trusted_ip_list.append(remote.ip_address)
+
+    ips = ','.join(trusted_ip_list)
+    return ips
+
+
+def distribute_iscsi_gateway_cfg(ctx, conf_data):
+    """
+    Distribute common gateway config to get the IPs
+    """
+    log.info('Distributing iscsi-gateway.cfg...')
+    for remote, roles in ctx.cluster.remotes.items():
+        remote.write_file(
+            path='/etc/ceph/iscsi-gateway.cfg',
+            data=conf_data,
+            sudo=True)
```

Comment:
I don't get get this. you're deploy this to the cluster hosts, right? who is reading this config file?

Reply:
this is used by the iscsi client to extract trusted_ip_list mainly: https://github.com/ideepika/ceph/blame/69155e8b990a92c1e7af4b7221747dc4ea28ccfa/src/test/cli-integration/rbd/iscsi_client.t#L3